generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/NumericValueStep.java

```diff
@@ -19,4 +27,51 @@ public void whenFieldIsConstrainedByNumericValue(String fieldName, String constr
     public void whenFieldIsNotConstrainedByNumericValue(String fieldName, String constraintName, Number value) {
         this.state.addNotConstraint(fieldName, constraintName, value);
     }
+
+    @Then("{fieldVar} contains numeric data")
+    public void producedDataShouldContainNumericValuesForField(String fieldName){
+        this.helper.assertFieldContainsNullOrMatching(fieldName, Number.class);
+    }
+
+    @Then("{fieldVar} contains numeric values between {number} and {number} inclusively")
+    public void producedDataShouldContainNumericValuesInRangeForField(String fieldName, Number minInclusive, Number maxInclusive){
+        this.helper.assertFieldContainsNullOrMatching(
+            fieldName,
+            Number.class,
+            value -> isGreaterThanOrEqual(value, minInclusive) && isLessThanOrEqual(value, maxInclusive));
+    }
+
+    private boolean isGreaterThanOrEqual(Number value, Number minInclusive){
+        BigDecimal valueAsBigDecimal = coerceToBigDecimal(value);
+        BigDecimal minInclusiveAsBigDecimal = coerceToBigDecimal(minInclusive);
+
+        return valueAsBigDecimal.compareTo(minInclusiveAsBigDecimal) >= 0;
+    }
+
+    private boolean isLessThanOrEqual(Number value, Number maxInclusive){
+        BigDecimal valueAsBigDecimal = coerceToBigDecimal(value);
+        BigDecimal maxInclusiveAsBigDecimal = coerceToBigDecimal(maxInclusive);
+
+        return valueAsBigDecimal.compareTo(maxInclusiveAsBigDecimal) <= 0;
+    }
+
+    private BigDecimal coerceToBigDecimal(Number number){
```

Comment:
I feel like I've seen this exact method somewhere else, can we refactor to use this method here?